Fix Gradle tests cache

idyatlov opened the pull request:
Test cache wasn't working (probably even after the upgrade to Ktor 2.x) because it started to send the data in chunked format without Content-Length - it resulted in `411 Length Required` response from build cache node.

It wasn't caught because for integration tests we use `SimpleEntryWriter` which probably contains not so much of data as standard `TestResultEntryWriter` and could be sent in one chunk, and there wasn't any logging in case of non-success responses.

One of the solutions was listed [here](https://slack-chats.kotlinlang.org/t/460857/i-am-trying-to-upgrade-our-code-to-ktor-client-2-0-2-but-get#b7ff417a-1d9b-479e-9613-ddc6f578df86) - to use `ByteArrayContent` with some length and it worked well. Probably it's worth to use `DataInputStream` for `CacheReader` as well to have uniform APIs, but it can be done separately because right now everything is working well.

technoir42 commented on core/src/main/kotlin/com/malinskiy/marathon/cache/test/serialization/TestResultEntryWriter.kt:
```diff
@@ -4,14 +4,14 @@ import com.malinskiy.marathon.cache.CacheEntryWriter
 class TestResultEntryWriter(private val testResult: TestResult) : CacheEntryWriter {
 
-    override suspend fun writeTo(output: ByteWriteChannel) {
+    override fun writeTo(output: DataOutputStream) {
```
From API perspective, I think this should accept the most generic type - `OutputStream`, but create the more specific type - `DataOutputStream` internally to use `writeInt`, `writeLong`, etc.

Commit:
Fix storing cache entities

Use ByteArrayContent instead of WriteChannelContent when executing HTTP PUT requests

technoir42 commented on core/src/main/kotlin/com/malinskiy/marathon/cache/test/serialization/TestResultEntryWriter.kt:
```diff
@@ -4,45 +4,42 @@ import com.malinskiy.marathon.cache.CacheEntryWriter
-        output.writeString(testResult.batchId)
-        output.writeString(testResult.stacktrace)
+    override fun writeTo(output: OutputStream) {
+        DataOutputStream(output).use {
```
Is it expected that we allow only one `writeTo` call to the provided stream? I wouldn't close it

Commit:
Don't close the stream in writeTo
